.github/workflows/python-app.yml

```
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
```

```suggestion
          pip install flake8==1.2.3 pytest=1.2.3
```

or at least specifcy minor or major version.... a bugfix release is ok to be automatically used,


```suggestion
          pip install flake8==1 pytest=1.2
```

------------

For this project it doesn't really matter... but when you have a big project you are working on a new change and then start to see unrelated failures generated by an auto-update of the dependencies.

When we update something, we have a separate feature branch in which we just do the update and make sure all tests are green :)